bvm/ballerina-core/src/main/java/org/ballerinalang/persistence/serializable/serializer/providers/instance/SerializedKeyInstanceProvider.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * you may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.ballerinalang.persistence.serializable.serializer.providers.instance;
+
+import org.ballerinalang.model.util.serializer.TypeInstanceProvider;
+import org.ballerinalang.persistence.serializable.SerializedKey;
+
+/**
+ * Provide object instance to represent {@link SerializedKey}.
+ *
+ * @since 0.98.1
+ */
+public class SerializedKeyInstanceProvider implements TypeInstanceProvider {
+    @Override
+    public String getTypeName() {
+        return SerializedKey.class.getName();
+    }
+
+    @Override
+    public Object newInstance() {
+        return new SerializedKey(null);
```

Comment:
Why passing a `null`?

Reply:
Only requirement of newInstance() method is to give us a instance of that particular type.
That instance will be populated to proper state later in the deserialization process by overwriting all the fields with target values.

Reply:
But this can cause NPE later, in downstream code. Shall we pass a proper value?